framework/src/wifi_manager/wifi_utils.c

```diff
@@ -0,0 +1,466 @@
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA_AND_WPA2_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA2_MIXED) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA2_CCMP) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA2_TKIP) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA2_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA_MIXED) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP_AND_AES;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA_CCMP) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_AES;
+			} else if (results->sec_modes->secmode == SLSI_SEC_MODE_WPA_TKIP) {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_WPA_PSK;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_TKIP;
+			} else {
+				ap_list[bss_count].ap_auth_type = WIFI_UTILS_AUTH_UNKNOWN;
+				ap_list[bss_count].ap_crypto_type = WIFI_UTILS_CRYPTO_UNKNOWN;
+			}
+		}
+
+		results = results->next;
+		if (ap_list[bss_count].ap_auth_type != WIFI_UTILS_AUTH_UNKNOWN && ap_list[bss_count].ap_crypto_type != WIFI_UTILS_CRYPTO_UNKNOWN) {
+			bss_count++;
+		}
+	} while (results->next && bss_count < list_size);
+
+	*found_ap_count = bss_count;
+	results = head;
+	WiFiFreeScanResults(&results);
+
+	int i;
+	for (i = 0; i < *found_ap_count && i < list_size; i++) {
+		int j;
+		int max = i;
+		for (j = i + 1; j < *found_ap_count && i < list_size; j++) {
+			if (ap_list[max].rssi < ap_list[j].rssi) {
+				max = j;
+			}
+		}
+
+		wifi_utils_ap_scan_info tmp;
+		tmp = ap_list[i];
+		ap_list[i] = ap_list[max];
+		ap_list[max] = tmp;
+	}
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+wifi_utils_result wifi_utils_register_connection_callback(void *connect_event_func, void *disconnect_event_func)
+{
+	g_linkup = connect_event_func;
+	g_linkdown = disconnect_event_func;
+
+	return WIFI_UTILS_SUCCESS;
+}
+
+wifi_utils_result wifi_utils_connect_ap(wifi_utils_ap_connect_config *ap_connect_config)
+{
+	if (!ap_connect_config) {
+		return WIFI_UTILS_INVALID_ARGS;
+	}
+
+	int ret;
+	wifi_utils_result result = WIFI_UTILS_FAIL;
+	slsi_security_config_t *config = NULL;
+
+	if (ap_connect_config->passphrase && ap_connect_config->passphrase_length) {
+		config = (slsi_security_config_t *)zalloc(sizeof(slsi_security_config_t));
+		if (!config) {
+			ndbg("Memory allocation failed!\n");
+			goto connect_ap_fail;
```

Comment:
In this case, we don't need to check config so that let's change this to "return result;"